### PR TITLE
feat: dataset editor improvement

### PR DESCRIPTION
**Pull request by lilykuang**
### SUMMARY

- move add item button to top
- increase the size of SQL Expression
- add placeholder text 
- change x to delete icon 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

After:
Screen Shot 2020-07-28 at 10 31 26 AM
Screen Shot 2020-07-27 at 3 45 59 PM
Screen Shot 2020-07-28 at 10 40 48 AM

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/datasource/DatasourceEditor.jsx [hunk 1]
```diff
@@ -83,18 +83,23 @@ function ColumnCollectionTable({
               <Field
                 fieldKey="expression"
                 label={t('SQL Expression')}
-                control={<TextControl />}
+                control={
+                  <TextAreaControl
```
Comment (mistercrunch) [inline superset-frontend/src/datasource/DatasourceEditor.jsx:87]: That looked pretty big in the <tr>, would look bad if there was many of these stacked... We may want to inline it.

Comment (lilykuang) [reply superset-frontend/src/datasource/DatasourceEditor.jsx:87]: update the sql expression field to half the size

**Review by willbarrett — APPROVED** [review]
LGTM - if it LGT @nytai let's merge.

**Review by nytai — APPROVED** [review]
lgtm

**Pull request merged**